core/shared/platform/windows/platform_internal.h

```diff
@@ -137,14 +136,46 @@ bh_atomic_thread_fence(int mem_order);
 
 #define os_atomic_thread_fence bh_atomic_thread_fence
 
-typedef HANDLE os_file_handle;
-typedef void *os_dir_stream;
+typedef enum windows_handle_type {
+    windows_handle_type_socket,
+    windows_handle_type_file
+} windows_handle_type;
+
+typedef enum windows_access_mode {
+    windows_access_mode_read = 1 << 0,
+    windows_access_mode_write = 1 << 1
+} windows_access_mode;
+
+typedef struct windows_handle {
+    windows_handle_type type;
+    windows_access_mode access_mode;
+    union {
+        HANDLE handle;
+        SOCKET socket;
+    } raw;
+} windows_handle;
+
+typedef struct windows_dir_stream {
+    // Enough space for the wide filename and the info struct itself
+    char info_buf[PATH_MAX * sizeof(wchar_t) + sizeof(FILE_ID_BOTH_DIR_INFO)];
+    char current_entry_name[PATH_MAX];
+    // An offset into info_buf to read the next entry from
+    DWORD cursor;
+    int cookie;
+    windows_handle *handle;
+} windows_dir_stream;
+
+typedef windows_handle *os_file_handle;
+typedef windows_dir_stream *os_dir_stream;
+
 #if WASM_ENABLE_UVWASI != 1
 typedef HANDLE os_raw_file_handle;
 #else
 typedef uint32_t os_raw_file_handle;
 #endif
 
+#define bh_socket_t os_file_handle
```

Comment:
why not typedef?

Reply:
Agree it would be better to typedef but it's defined as a macro on other platforms and if I define it via typedef, I get a bunch of errors due to [this](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/include/platform_common.h#L115-L119) which overwrites the type as`int`. It could be fixed but then I'd need to touch all the other `platform_internal.h`'s which don't define `bh_socket_t` and add a definition so we can remove the default macro define in platform_common.h.

Actually, we can completely remove the `bh_socket_t` type since it always has to be the same as `os_file_handle`. I can raise another PR later on to fix that as part of the Windows work if that sounds good?